### DISCUSSION TITLE
How does @babel/preset-env transpile the JS code based on Browserslist?

### DISCUSSION THREAD

**Discussion by mrdulin** [browserslist/browserslist#835] [Q&A]
For example, the `"chrome: 49"` query. My understand is babel will transpiled the JS code compatible with Chrome 49 version.

How does `@babel/preset-env` know what kind of transpiled JS code should be emited? What babel plugin, polyfill should be used?

The query is very fragmented, and it seems impossible to use a mapping of query and plugins, so babel can get the needed plugins by query.

For example：

- babel will include plugin-a, plugin-b, polyfill-a, polyfill-b for `"chrome 49"` query.
- babel will include plugin-a, polyfill-a for `"chrome 86"` query.

I didn't find the source code or any doc explain this process.

**Comment by ai** [comment]
You should ask Babel. We only transpile high level queries like `last 2 versions` to exact list of versions. 

But as I know they use some database of version>feature.

**Reply by mrdulin** [reply]
Can you explain "transpile high level queries like last 2 versions to exact list of versions."

What's the exact list of versions?
